aptos-move/framework/aptos-framework/sources/jwks.move

```diff
@@ -0,0 +1,480 @@
+/// JWK functions and structs.
+module aptos_framework::jwks {
+    use std::error::invalid_argument;
+    use std::option;
+    use std::option::Option;
+    use std::string::String;
+    use std::vector;
+    use aptos_std::copyable_any::{Any, pack};
+    use aptos_framework::event::emit;
+    use aptos_framework::reconfiguration;
+    use aptos_framework::system_addresses;
+    #[test_only]
+    use std::string;
+    #[test_only]
+    use std::string::utf8;
+    #[test_only]
+    use aptos_std::comparator::{compare_u8_vector, is_greater_than, is_equal};
+    #[test_only]
+    use aptos_std::copyable_any;
+
+    friend aptos_framework::genesis;
+
+    const EUNEXPECTED_EPOCH: u64 = 1;
+    const EUNEXPECTED_VERSION: u64 = 2;
+    const EUNKNOWN_JWK_MAP_EDIT: u64 = 3;
+
+    /// An OIDC provider.
+    struct OIDCProvider has drop, store {
+        name: vector<u8>, // e.g., b"https://www.facebook.com"
+        config_url: vector<u8>, // e.g., b"https://www.facebook.com/.well-known/openid-configuration/"
+    }
+
+    /// The OIDC provider set. Maintained by governance proposals.
+    struct OIDCProviderSet has key {
+        providers: vector<OIDCProvider>, // The order does not matter.
+    }
+
+    /// Some extra configs that controls JWK consensus behavior. Maintained by governance proposals.
+    ///
+    /// Currently supported `content` types:
+    /// - `JWKConsensusConfigV0`
+    struct JWKConsensusConfig has drop, key {
+        content: Any,
+    }
+
+
+    struct JWKConsensusConfigV0 has copy, drop, store {
+        observation_interval_ms: u64,
+    }
+
+    /// An observed but not yet supported JWK.
+    struct UnsupportedJWK has copy, drop, store {
+        payload: vector<u8>,
+    }
+
+    /// A JWK where `kty` is `RSA`.
+    struct RSA_JWK has copy, drop, store {
+        kid: String,
+        kty: String,
+        alg: String,
+        e: String,
+        n: String,
+    }
+
+    /// A JWK.
+    ///
+    /// Currently supported `content` types:
+    /// - `RSA_JWK`
+    /// - `UnsupportedJWK`
+    struct JWK has copy, drop, store {
+        id: vector<u8>,
+        content: Any,
```

Comment:
could you add a comment for content too? especially since type is Any

maybe note that Any is for example RSA_JWK